src/Centreon/Application/Controller/MonitoringResourceController.php

```diff
@@ -74,22 +70,62 @@ public function __construct(ResourceServiceInterface $resource)
     /**
      * List all the resources in real-time monitoring : hosts and services.
      *
-     * @param \Centreon\Domain\RequestParameters\Interfaces\RequestParametersInterface $requestParameters
+     * @param RequestParametersInterface $requestParameters
+     * @param Request $request
+     * @param SerializerInterface $serializer
+     * @param EntityValidator $entityValidator
      * @return View
      */
-    public function list(RequestParametersInterface $requestParameters): View
-    {
+    public function list(
+        RequestParametersInterface $requestParameters,
+        Request $request,
+        SerializerInterface $serializer,
+        EntityValidator $entityValidator
+    ): View {
+        // ACL check
         $this->denyAccessUnlessGrantedForApiRealtime();
 
-        $filterState = [];
-        foreach ($this->parseExtraParameter($requestParameters, static::EXTRA_PARAMETER_STATE) as $state) {
-            if (!in_array($state, ResourceService::STATES)) {
+        // set default values of filter data
+        $filterData = [];
+        foreach (static::EXTRA_PARAMETERS_LIST as $param) {
+            $filterData[$param] = [];
+        }
+
+        // load filter data with the query parameters
+        foreach ($request->query as $param => $data) {
+            // skip pagination parameters
+            if (in_array($param, ['search', 'limit', 'page', 'sort_by'])) {
                 continue;
             }
 
-            $filterState[] = $state;
+            try {
+                $value = json_decode($data, true);
```

Comment:
By default, json_decode does not throw exception.
IMHO you should check if $value is null and throw an InvalidArgumentException.
And do not affect $data to $value is it's not a correct json

Reply:
I forgot to add JSON_THROW_ON_ERROR parameter